greedy_algorithms/DJnew.cpp

```diff
@@ -0,0 +1,149 @@
+        std::vector<int> distance;
+        distance.resize(graph.numVertices);
+        std::vector<int> prev;
+        prev.resize(graph.numVertices);
+
+        for (int v = 0; v < distance.size(); v++) {
+            distance[v] = 99999;
+            prev[v] = -1;
+            Q.insert(v);
+        }
+        distance[src] = 0;
+
+        while (!Q.empty()) {
+            int u = *Q.begin();
+            for (int i : Q) {
+                //check if distance is greater than...
+                if (distance[i] < distance[u]) { // checks the distance instead of vertex
+                    u = i;
+                }
+            }
+            Q.erase(u);
+
+            std::vector<std::pair<int, int>> neighbors = graph.adjList[u];
+            for (std::pair<int, int> n : neighbors) {
+                //check if it is in Q
+                if (Q.find(n.first) != Q.end()) {
+                    if (distance[u] + n.second < distance[n.first]) {
+                        distance[n.first] = distance[u] + n.second;
+                        prev[n.first] = u;
+                    }
+                }
+            }
+        }
+
+        return distance;
+    }
+
+/**
+ * @brief Self-test implementations
+ * @returns void
+ */
+
+    static void test() {
+        std::vector<Edge> edges1 =
+                {
+                    {0,0, 1}, {0,1, 2}, {1,2, 0}, {1,2, 1}, {2,3, 2}, {3,4, 5}, {3,5, 4}
+                };
+        Graph obj1(edges1 , 6);
+        std::vector<int> test1Key{ 0,2,2,4,9,8};
+        bool result = std::equal(dijkstra(obj1,0).begin(), dijkstra(obj1,0).end(), test1Key.begin());
+        if (result)
+            std::cout << "passed";
+        //0 2 2 4 9 8
+
+        std::cout<<std::endl;
+        std::vector<Edge> edges2 =
+                {
+                        {0,1, 3}, {1,2, 5}, {2,3, 3}, {2,3, 6}, {3,4, 2}, {4,3, 4}, {4,5, 3}
+                };
+        Graph obj2(edges2, 6);
+        //0 3 8 11 13 16
+        std::vector<int> test2Key{0,3,8,11,13,16};
+        bool result2 = std::equal(dijkstra(obj2,0).begin(), dijkstra(obj2,0).end(), test2Key.begin());
+        if (result2)
+            std::cout << "passed";
+        std::cout<<std::endl;
+        //test 3
+        std::vector<Edge> edges3 =
+                {
+                        {0,1, 4}, {1,2, 6}, {2,3, 3}, {2,5, 6}, {3,4, 2}, {4,2, 2}, {4,4, 4}
+                };
+        Graph obj3(edges3 , 6);
+        std::vector<int> test3Key{ 0, 4, 10, 13, 15, 16};
+        //0 4 10 13 15 16
+        bool result3 = std::equal(dijkstra(obj3,0).begin(), dijkstra(obj3,0).end(), test3Key.begin());
+        if (result3)
+            std::cout << "passed" ;
+
+    }
+}
+/**
+ * @brief Main function
+ * @param argc commandline argument count (ignored)
+ * @param argv commandline array of arguments (ignored)
+ * @returns 0 on exit
+ */
+int main() {
+    greedy_algorithms::test();  // run self-test implementations
+    // code here
+    return 0;
+}
```

Comment:
Missing an endline. Without this endline, there could be issues with some IDEs and configurations.
```suggestion
}

```